README.md

```diff
@@ -1 +1,253 @@
+
+      while (1) {
+        preamble_time = (off_time + (2 * on_time)) / 1000000;
+        preamle_bit_length = (preamble_time * bit_rate);
+        if (preamle_bit_length <= 50000) {
+          break;
+        }
+        off_time = off_time - on_time;
+      }
+
+      if (((uint32_t)off_time) != DUTY_CYCLE_OFF_TIME) {
+        app_log_warning("Duty Cycle Off time was changed to ensure stable working\n");
+      }
+
+      return preamle_bit_length;
+    }
+    ```
+### Modify the Radio PHY configuations
+Please download the ***rail_config.c*** and ***rail_config.h*** file as the following link, and replace the original file is located in **Autogen** folder, rebuild the project and flash to the device.   
+[Optimized 9p6k radio configuation](radio_configuations/optimized_9p6k_configuation)  
+
+**Note**: Those two files will be overwritten if you change anythings in radio generator.  
+
+## Test conclusion
+### Power Consumption
+After optimzing the EM2 current and radio PHY. **Iavg = 19.036 uA**  
+<img src="images/optimized power consumption.gif">
+
+Below table is the comparison of power consumption between the sample project and the optimized project.
+|   project   |sample peoject| optimized project| 
+|:----:| :-----------:| :----------------:| 
+|**average current**| 40.674 uA| 19.036 uA |  
+
+**Conclusion:** The power consumption reduce by **53%** to **19.036 uA** after optimizing. 
+### Sensitivity
+- Rx project: Optimized long preamble duty cycle
+-  Use the [LR_Waveform_Generator](https://github.com/silabs-JimL/LR_WaveFormGenerator) to generate a waveform file and download the E4432B siganl generator. When the PER is 1%, the value at the monment represent the sensiticity.
+-  Sensitivity is **-118.4 dBm** after optimizing. Below table shows the test data.  
+  
+    |  output power    |Tx packets| Rx packets| 
+    |:----:| :-----------:| :----------------:| 
+    |-119.2 dBm| 1000 | 974 |
+    |-118.8 dBm| 1000 | 982 |
+    |-118.6 dBm| 1000 | 989 |
+    |-118.5 dBm| 1000 | 987 | 
+    |**-118.4 dBm**| **1000** | **991** |
+    |-118.3 dBm| 1000 | 994 |
+
+-  The sensitivity of original long preamble duty cycle is **-119.2 dBm**. Below table shows the test data.
+    |  output power    |Tx packets| Rx packets| 
+    |:----:| :-----------:| :----------------:| 
+    |-119.8 dBm| 1000 | 981 |
+    |-119.6 dBm| 1000 | 987 |
+    |-119.4 dBm| 1000 | 984 |
+    |**-119.2 dBm**| **1000** | **993** | 
+    |-119.1 dBm| 1000 | 995 |
+   
+-  Below table is the comparison of sensitivity between the sample project and the optimized project.     
+
+    |   project   |sample peoject| optimized project| 
+    |:----:| :-----------:| :----------------| 
+    |**sensitivity**| -119.2 dBm| -118.4 dBm|    
+
+  **Conclusion:** Compared with original example, there is a **0.8 dBm** loss on sensitivity. But the loss is trivial and the optimization is well worth to do it.  
```

Comment:
The sensitivity of the sample project with optimized radio configuration reduces about **0.8 dBm**, which is insignificant and can be accepted by customers in order to get better power consumption.